### PR TITLE
MINOR: fix javadoc warnings

### PR DESCRIPTION
**Pull request by gaurav-narula**
Fixes javadoc warning for unescaped `<` and `>` chars.

### CODE REVIEW

File: core/src/main/java/kafka/log/remote/RemoteLogManager.java [hunk 1]
```diff
@@ -1277,7 +1277,8 @@ public static boolean isRemoteSegmentWithinLeaderEpochs(RemoteLogSegmentMetadata
      * does not contain any messages/records associated with them.
      *
      * For ex:
-     *  <epoch - start offset>
+     * <pre>
+     *  &lt;epoch - start offset&gt;
```
Comment (nikramakrishnan) [inline core/src/main/java/kafka/log/remote/RemoteLogManager.java]: Does it make sense to cover this and other 'tables' in `@code` like:

```java
* <pre>
* {@code
* <epoch - start offset>
* ...
* }
* </pre>
```

to allow for readability even when reading the class file directly?

**Review by nikramakrishnan — COMMENTED** [review]
Thanks for the change! Left a minor comment.

File: core/src/main/java/kafka/log/remote/RemoteLogManager.java (hunk 1 above)
Comment (gaurav-narula) [reply core/src/main/java/kafka/log/remote/RemoteLogManager.java]: Addressed in [01e23fb](https://github.com/apache/kafka/pull/15527/commits/01e23fb2520338bcd49a69da478fea37859de00a)

**Review by nikramakrishnan — APPROVED** [review]
Thanks for this change, LGTM!

**Comment by gharris1727** [comment]
Hi @gaurav-narula I see some other javadoc warnings, do you think we should address these?

```
> Task :storage:storage-api:javadoc
kafka/storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteStorageMetrics.java:31: warning: reference not found: kafka.api.MetricsTest
 * @see kafka.api.MetricsTest
   ^
1 warning

> Task :streams:javadoc
kafka/streams/src/main/java/org/apache/kafka/streams/state/QueryableStoreTypes.java:52: warning: invalid input: '<'
     * A {@link QueryableStoreType} that accepts {@link ReadOnlyKeyValueStore ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>>}.
                                                                                                   ^
kafka/streams/src/main/java/org/apache/kafka/streams/state/QueryableStoreTypes.java:74: warning: invalid input: '<'
     * A {@link QueryableStoreType} that accepts {@link ReadOnlyWindowStore ReadOnlyWindowStore<K, ValueAndTimestamp<V>>}.
                                                                                               ^
kafka/streams/src/main/java/org/apache/kafka/streams/state/QueryableStoreTypes.java:52: warning: invalid input: '<'
     * A {@link QueryableStoreType} that accepts {@link ReadOnlyKeyValueStore ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>>}.
                                                                                                   ^
kafka/streams/src/main/java/org/apache/kafka/streams/state/QueryableStoreTypes.java:74: warning: invalid input: '<'
     * A {@link QueryableStoreType} that accepts {@link ReadOnlyWindowStore ReadOnlyWindowStore<K, ValueAndTimestamp<V>>}.
                                                                                               ^
kafka/streams/src/main/java/org/apache/kafka/streams/state/QueryableStoreTypes.java:52: warning: invalid input: '<'
     * A {@link QueryableStoreType} that accepts {@link ReadOnlyKeyValueStore ReadOnlyKeyValueStore<K, ValueAndTimestamp<V>>}.
                                                                                                   ^
kafka/streams/src/main/java/org/apache/kafka/streams/state/QueryableStoreTypes.java:74: warning: invalid input: '<'
     * A {@link QueryableStoreType} that accepts {@link ReadOnlyWindowStore ReadOnlyWindowStore<K, ValueAndTimestamp<V>>}.
```

**Comment by gaurav-narula** [comment]
@gharris1727 Thanks for pointing that out.

The warning below
```
kafka/storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteStorageMetrics.java:31: warning: reference not found: kafka.api.MetricsTest
 * @see kafka.api.MetricsTest
```

occurs because we're trying to refer to a class in the test compile path from production code. I don't think that's a good idea. Usually IDEs help in figuring out the usages of fields/classes anyway. Should we remove that tag? WDYT?

**Comment by gharris1727** [comment]
> I don't think that's a good idea. Usually IDEs help in figuring out the usages of fields/classes anyway. Should we remove that tag?

I agree, let's remove that. RemoteStorageMetrics is already referenced by MetricsTest, so this back-link isn't adding anything.

**Comment by gaurav-narula** [comment]
@gharris1727 I've addressed the feedback. Please have a look when you get some time.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**